### PR TITLE
update(server): Adding IsPistolRound()

### PR DESCRIPTION
**Pull request by blu133721**
Adding IsPistolRound().

### CODE REVIEW

File: plugin_files/scripting/includes/swiftly/server.h [hunk 1]
```diff
@@ -50,6 +52,11 @@ class Server
         this->ExecuteCommand("changelevel %s", map);
     }
 
+    bool IsPistolRound(bool *g_bPistolRound)
+    {
+        g_bPistolRound = g_pGameRules->m_totalRoundsPlayed() == 0 || (g_pGameRules->m_bSwitchingTeamsAtRoundReset() && g_pGameRules->m_nOvertimePlaying() == 0) || g_pGameRules->m_bGameRestart();
```
Comment (skuzzis) [inline plugin_files/scripting/includes/swiftly/server.h:57]: bro? you forgot you need to use FetchFunctionPtr?

File: src/components/Plugins/src/language/lua/server.cpp [hunk 2]
```diff
@@ -27,8 +27,9 @@ void SetupLuaServer(luacpp::LuaState *state, Plugin *plugin)
                    { scripting_Server_ExecuteCommand(command); })
         .DefMember("ChangeLevel", [](LuaServerClass *base, const char *map) -> void
                    {
-            if(!scripting_Server_IsMapValid(map)) return;
-            scripting_Server_ExecuteCommand(string_format("map %s", map).c_str()); });
-
-    state->DoString("server = Server()");
```
Comment (skuzzis) [inline src/components/Plugins/src/language/lua/server.cpp:33]: where is this now?

**Review by skuzzis — CHANGES_REQUESTED** [review]

File: src/components/Plugins/src/language/lua/server.cpp (hunk 2 above)
Comment (blu133721) [reply src/components/Plugins/src/language/lua/server.cpp:33]: Fixed in the latest commit.

File: plugin_files/scripting/includes/swiftly/swiftly_memory.h [hunk 3]
```diff
@@ -75,6 +75,7 @@ typedef const char *(*Server_GetMap)();
 typedef uint16_t (*Server_GetMaxPlayers)();
 typedef bool (*Server_IsMapValid)(const char *);
 typedef void (*Server_AddPrecacheModel)(const char *);
+typedef bool (*Server_IsPistolRound)(bool);
```
Comment (skuzzis) [inline plugin_files/scripting/includes/swiftly/swiftly_memory.h:78]: Remove bool from argument

**Review by skuzzis — CHANGES_REQUESTED** [review]

File: plugin_files/scripting/includes/swiftly/swiftly_memory.h (hunk 3 above)
Comment (blu133721) [reply plugin_files/scripting/includes/swiftly/swiftly_memory.h]: Just removed it.

File: plugin_files/scripting/includes/swiftly/server.h [hunk 4]
```diff
@@ -50,6 +52,18 @@ class Server
         this->ExecuteCommand("changelevel %s", map);
     }
 
+    bool IsPistolRound(bool *g_bPistolRound)
```
Comment (skuzzis) [inline plugin_files/scripting/includes/swiftly/server.h:55]: argument?

**Review by skuzzis — CHANGES_REQUESTED** [review]

**Review by skuzzis — APPROVED** [review]

**Comment by skuzzis** [comment]
The PR will be pushed once the checks will be completed.

**Pull request merged**